test/extended/networking/multinetpolicy.go

```diff
@@ -0,0 +1,102 @@
+
+	v1 "k8s.io/api/core/v1"
+	frameworkpod "k8s.io/kubernetes/test/e2e/framework/pod"
+
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[sig-network][Feature:MultiNetworkPolicy]", func() {
+	var oc *exutil.CLI
+	var ns string
+
+	oc = exutil.NewCLIWithPodSecurityLevel("multinetpol-e2e", admissionapi.LevelBaseline)
+
+	f := oc.KubeFramework()
+	podName := "multinetpol-test-pod-"
+
+	g.It("should enforce a network policies on secondary network", func() {
+		var err error
+		ns = f.Namespace.Name
+
+		g.By("creating a macvlan net-attach-def")
+		nad_yaml := exutil.FixturePath("testdata", "net-attach-defs", "macvlan-nad.yml")
+		err = oc.AsAdmin().Run("create").Args("-f", nad_yaml).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("launching pod with an annotation to use the net-attach-def")
+		networksTemplate := `[
+			  {
+				"name": "macvlan1-nad",		
+				"ips": ["%s"]
+			  }
+			]`
+
+		frameworkpod.CreateExecPodOrFail(f.ClientSet, ns, podName, func(pod *v1.Pod) {
+			pod.Spec.Containers[0].Args = []string{"net", "--serve", "2.2.2.1:8889"}
+			pod.ObjectMeta.Labels = map[string]string{"pod": "a"}
+			pod.ObjectMeta.Annotations = map[string]string{
+				"k8s.v1.cni.cncf.io/networks": fmt.Sprintf(networksTemplate, "2.2.2.1/24")}
+		})
+
+		testPodB := frameworkpod.CreateExecPodOrFail(f.ClientSet, ns, podName, func(pod *v1.Pod) {
+			pod.ObjectMeta.Labels = map[string]string{"pod": "b"}
+			pod.ObjectMeta.Annotations = map[string]string{
+				"k8s.v1.cni.cncf.io/networks": fmt.Sprintf(networksTemplate, "2.2.2.2/24")}
+		})
+
+		g.By("checking podB can connnect to podA")
+		output, err := oc.AsAdmin().Run("exec").Args(testPodB.Name, "--", "curl", "--connect-timeout", "1", "2.2.2.1:8889").Output()
+		o.Expect(err).To(o.Succeed(), "cmd output: %s", output)
+
+		g.By("enabling MultiNetworkPolicies on cluster")
+		enableMultiNetworkPolicy(oc)
+		defer disablMultiNetworkPolicy(oc)
+
+		g.By("creating a deny-all-traffic MultiNetworkPolicy")
+		multinetpolicy_yaml := exutil.FixturePath("testdata", "multinetpolicy", "deny-all.yml")
+		err = oc.AsAdmin().Run("create").Args("-f", multinetpolicy_yaml).Execute()
+		o.Expect(err).To(o.Succeed())
+
+		g.By("checking podB can NOT connnect to podA")
+		o.Eventually(func() error {
+			_, err = oc.AsAdmin().Run("exec").Args(testPodB.Name, "--", "curl", "--connect-timeout", "1", "2.2.2.1:8889").Output()
+			return err
+		}, "30s", "1s").Should(o.HaveOccurred())
+	})
+
+})
+
+func enableMultiNetworkPolicy(oc *exutil.CLI) {
+	err := oc.AsAdmin().
+		WithoutNamespace().
+		Run("patch").
+		Args("network.operator.openshift.io", "cluster", "--type=merge", "--patch", `{"spec":{"useMultiNetworkPolicy": true}}`).
+		Execute()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	o.Eventually(func() error {
+		return oc.AsAdmin().Run("get").Args("multi-networkpolicies.k8s.cni.cncf.io").Execute()
+	}, "30s", "2s").Should(o.Succeed())
```

Comment:
This doesn't reliably indicate that MNP _enforcement_ is in effect... that depends on deploying a DaemonSet as well, right?

The timeouts help, but it would be better if there was a good way of ensuring that the DaemonSet is fully deployed, so the test doesn't end up flaking under heavy load. (But also, we don't want to make too many assumptions about exactly what CNO is doing when you enable the feature, especially since we might change to a different MNP implementation in the future... I'm not sure what the best approach here is...)

Reply:
I agree with your considerations, it's not easy to decide what's the best approach. From the user's perspective, turning on the feature means that MNP resource becomes available. 

If the Daemonset (or any other component) is not working or it is not started, then some other assertion will fail.

As a further example, in the future we can also decide to keep the Daemonset up and running but not creating iptables if the flag is set to false.

That said, I would keep it as simple as possible.